object/id/id.go

```diff
@@ -134,22 +134,16 @@ func (id ID) String() string {
 
 // CalculateIDSignature signs object id with provided key.
 func (id ID) CalculateIDSignature(signer neofscrypto.Signer) (neofscrypto.Signature, error) {
-	data, err := id.Marshal()
-	if err != nil {
-		return neofscrypto.Signature{}, fmt.Errorf("marshal ID: %w", err)
-	}
-
 	var sig neofscrypto.Signature
-
-	return sig, sig.Calculate(signer, data)
+	return sig, sig.Calculate(signer, id.Marshal())
 }
 
 // Marshal marshals ID into a protobuf binary form.
-func (id ID) Marshal() ([]byte, error) {
+func (id ID) Marshal() []byte {
```

Comment:
want this change but i am completely broken at understanding what we support (but no release still) and what we promise. it is exported and it is broken now (but some other suggestions have been already reverted before because of real users)

Reply:
`Marshal` is very rarely used accross all types by non-system apps (mostly it's called by things like `Sign/Verify/WriteSmthToSmth`). We desire to _not_ have error return from the function which cannot fail. While there is an opportunity before release, I want to achieve the desired look. Fortunately, even if the application code breaks, it will only become smaller

consider this as an exclusion. Historically all `Marshal`'s returned an error. Then we decided to drop error, but didn’t finish the job

Reply:
most of them are useless btw. For example, in `eacl` only `Table` is stably encoded. Other types have this method on a whim and no one interested in them